src/terminal/scroll.lua

```diff
@@ -109,17 +127,23 @@ function M.scroll_apply()
 end
 
 --- Pushes a new scroll region onto the stack without writing it.
--- @tparam[opt] number top Top row of the scroll region.
--- @tparam[opt] number bottom Bottom row of the scroll region.
+-- Negative indices are supported, counting from the bottom of the screen.
+-- @tparam[opt] number top Top row of the scroll region (can be negative).
+-- @tparam[opt] number bottom Bottom row of the scroll region (can be negative).
 -- @treturn string ANSI sequence of the new scroll region.
 function M.scroll_pushs(top, bottom)
+  if top and bottom then
+    top = resolve_index(top, get_height())
+    bottom = resolve_index(bottom, get_height())
+  end
```

Comment:
same again; this work should already be done in `scroll_regions`, so no need to do it again here.